File: problems/001/001.py
import sys
import time


def step_sum(start, end, step):
    (s, e) = (start // step, (end - 1) // step)

    return step * (e * (e + 1) / 2 + s * (s + 1) / 2)


def f():
    (s, e) = (0, 1000)

    return step_sum(s, e, 3) + step_sum(s, e, 5) - step_sum(s, e, 15)

if len(sys.argv) == 1:
    print(f())
elif len(sys.argv) == 2:
    iterations = int(sys.argv[1])

    start = time.time()
    for i in range(0, iterations):
        f()
    end = time.time()

    print(int((end - start) * 1000000000) // iterations)

File: problems/001/001.rs
extern crate test;
extern crate time;

use std::iter::AdditiveIterator;
use std::os::args;
use test::black_box;
use time::precise_time_ns;

#[inline]
fn f() -> u64 {
    range(0u64, 1000).filter(|x| x % 3 == 0 || x % 5 == 0).sum()
}

fn main() {
    match args().as_slice() {
        [_] => {
            println!("{}", f());
        },
        [_, ref iters] => {
            let iters: u64 = from_str(iters.as_slice()).unwrap();

            let start = precise_time_ns();
            for _ in range(0, iters) {
                black_box(f());
            }
            let end = precise_time_ns();

            println!("{}", end - start);
        },
        _ => unreachable!(),
    }
}
